Marlin/ultralcd.cpp

```diff
@@ -3961,22 +3964,29 @@ void lcd_update() {
   } // ELAPSED(ms, next_lcd_update_ms)
 }
 
-void set_utf_strlen(char* s, uint8_t n) {
-  uint8_t i = 0, j = 0;
-  while (s[i] && (j < n)) {
-    #if ENABLED(MAPPER_NON)
-      j++;
-    #else
-      if ((s[i] & 0xC0u) != 0x80u) j++;
-    #endif
-    i++;
+#if DISABLED(STATUS_MESSAGE_SCROLLING)
+
+  void set_utf_strlen(char* s, uint8_t n) {
+    uint8_t i = 0, j = 0;
+    while (s[i] && (j < n)) {
+      #if ENABLED(MAPPER_NON)
+        j++;
+      #else
+        if ((s[i] & 0xC0u) != 0x80u) j++;
+      #endif
+      i++;
+    }
+    while (j++ < n) s[i++] = ' ';
+    s[i] = '\0';
   }
-  while (j++ < n) s[i++] = ' ';
-  s[i] = '\0';
-}
+
+#endif // !STATUS_MESSAGE_SCROLLING
 
 void lcd_finishstatus(bool persist=false) {
-  set_utf_strlen(lcd_status_message, LCD_WIDTH);
+  #if DISABLED(STATUS_MESSAGE_SCROLLING)
```

Comment:
This might be why we're seeing the end of the last message being displayed when a shorter message is displayed. Prior to this change, `set_utf_strlen` always filled the `lcd_status_message` with spaces to ensure it was at least `LCD_WIDTH` characters long. However, now that `set_utf_strlen` is only being called if `STATUS_MESSAGE_SCROLLING` is disabled, messages shorter than `LCD_WIDTH` are not lengthened with spaces.